FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -20,29 +24,25 @@ public final class RemoteFeedLoader: FeedLoader {
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
 		client.get(from: url) { [weak self] result in
-			guard let self = self else { return }
+			guard self != nil else { return }
 
 			switch result {
 			case let .success((data, response)):
-				guard response.statusCode == 200 else {
-					return completion(.failure(Error.invalidData))
-				}
-
-				do {
-					let feedImages = try self.feedImages(from: data)
-					completion(.success(feedImages))
-				} catch {
-					completion(.failure(Error.invalidData))
-				}
+				completion(RemoteFeedLoader.feedLoaderResult(from: data, response: response))
 
 			case .failure:
 				completion(.failure(Error.connectivity))
 			}
 		}
 	}
 
-	private func feedImages(from data: Data) throws -> [FeedImage] {
-		let feedResult = try JSONDecoder().decode(FeedResult.self, from: data)
-		return feedResult.items.map { FeedImage($0) }
+	private static func feedLoaderResult(from data: Data, response: HTTPURLResponse) -> FeedLoader.Result {
+		guard
+			response.statusCode == Constants.okCode,
+			let feedResult = try? JSONDecoder().decode(FeedResult.self, from: data) else {
+			return .failure(Error.invalidData)
+		}
+
+		return .success(feedResult.items.map { FeedImage($0) })
```

Comment:
Suggestion: another way of using `map` in this situation is to pass the initializer as a parameter:
```suggestion
		return .success(feedResult.items.map(FeedImage.init))
```